src/modm/driver/pressure/ms5837_impl.hpp

```diff
@@ -0,0 +1,154 @@
+        RF_RETURN(false);
+    }
+
+
+    // 4 ms reload (?: from MS5611)
+    timeout.restart(std::chrono::milliseconds(4));
+    RF_WAIT_UNTIL(timeout.isExpired());
+
+    // Read the factory calibration from PROM
+    data.prom.data[0] = modm::fromBigEndian(RF_CALL(readProm(0)));
+    data.prom.data[1] = modm::fromBigEndian(RF_CALL(readProm(1)));
+    data.prom.data[2] = modm::fromBigEndian(RF_CALL(readProm(2)));
+    data.prom.data[3] = modm::fromBigEndian(RF_CALL(readProm(3)));
+    data.prom.data[4] = modm::fromBigEndian(RF_CALL(readProm(4)));
+    data.prom.data[5] = modm::fromBigEndian(RF_CALL(readProm(5)));
+    data.prom.data[6] = modm::fromBigEndian(RF_CALL(readProm(6)));
+    // data.prom.data[7] = modm::fromBigEndian(RF_CALL(readProm(7)));
+
+    MODM_LOG_DEBUG.printf("MS5837 Reset Initialize done\n");
+
+    RF_END_RETURN(data.prom.calculateCrc());
+}
+
+// -----------------------------------------------------------------------------
+
+template < typename I2cMaster >
+modm::ResumableResult<void>
+Ms5837<I2cMaster>::readout(OversamplingRatio osrPressure, OversamplingRatio osrTemperature)
+{
+    RF_BEGIN();
+
+    // start a pressure conversion
+    buffer[0] = i(Command::Convert) | i(Conversion::Pressure) | i(osrPressure);
+    this->transaction.configureWrite(buffer, 1);
+    if (!RF_CALL( this->runTransaction() )){
+        MODM_LOG_DEBUG.printf("MS5837 Pressure Conversion failed\n");
+        RF_RETURN();
+    }
+
+    // Wait until pressure conversion has finished
+    timeout.restart(std::chrono::milliseconds(conversionDelay[i(osrPressure) >> 1]));
+    RF_WAIT_UNTIL(timeout.isExpired());
+
+    // Get the pressure conversion result from sensor
+    buffer[0] = i(Command::AdcRead);
+
+    this->transaction.configureWriteRead(buffer, 1, data.raw, 3);
+    if (!RF_CALL( this->runTransaction() )){
+        MODM_LOG_DEBUG.printf("MS5837 Pressure Conversion failed\n");
+        RF_RETURN();
+    }
+
+    // Notify data class about changed buffer
+    data.rawPressureTouched();
+
+
+
+    // start a temperature conversion
+    buffer[0] = i(Command::Convert) | i(Conversion::Temperature) | i(osrTemperature);
+    this->transaction.configureWrite(buffer, 1);
+    if (!RF_CALL( this->runTransaction() )){
+        MODM_LOG_DEBUG.printf("MS5837 Temperature Conversion failed\n");
+        RF_RETURN();
+    }
+
+
+    // Wait until temperature conversion has finished
+    timeout.restart(std::chrono::milliseconds(conversionDelay[i(osrTemperature) >> 1]));
+    RF_WAIT_UNTIL(timeout.isExpired());
+
+    // Get the temperature conversion result from sensor
+    buffer[0] = i(Command::AdcRead);
+
+    this->transaction.configureWriteRead(buffer, 1, data.raw + 3, 3);
+    if (!RF_CALL( this->runTransaction() )){
+        MODM_LOG_DEBUG.printf("MS5837 Pressure Conversion failed\n");
+        RF_RETURN();
+    }
+
+    // Notify data class about changed buffer
+    data.rawTemperatureTouched();
+
+    RF_END();
```

Comment:
```suggestion
    RF_END_RETURN(true);
```
with the return type changes.